src/components/button/__tests__/Button.test.tsx

```diff
@@ -73,6 +73,37 @@ describe('components/button/Button', () => {
         expect(wrapper.prop('type')).toEqual('submit');
     });
 
+    test('should add bdl-is-large modifier class when size is "large"', () => {
+        const wrapper = shallow(<Button size="large">Click Here</Button>);
+
+        expect(wrapper.prop('className')).toEqual('btn bdl-is-large');
+    });
+
+    test('should render icon in icon container if icon prop is set', () => {
+        const FakeIcon = (props: Record<string, unknown>) => <svg {...props} />;
+        const wrapper = shallow(<Button icon={<FakeIcon />} />);
+        const iconContainer = wrapper.find('.bdl-btn-icon');
+        const { width, height } = iconContainer.find('FakeIcon').props();
+        expect(wrapper.props().className).toEqual('btn bdl-has-icon');
+        expect(iconContainer.length).toBe(1);
+        expect(width).toEqual(height);
+        expect(width).toEqual(20);
+    });
+
+    test('should render icon next to text if icon and children props are both set', () => {
+        const FakeIcon = (props: Record<string, unknown>) => <svg {...props} />;
+        const wrapper = shallow(<Button icon={<FakeIcon />}>Click Here</Button>);
+        const iconContainer = wrapper.find('.bdl-btn-icon');
+        const textContainer = wrapper.find('.btn-content');
+        const { width, height } = iconContainer.find('FakeIcon').props();
+        expect(iconContainer.length).toBe(1);
+        expect(textContainer.length).toBe(1);
+        expect(width).toEqual(height);
+        expect(width).toEqual(16);
+
+        expect(wrapper).toMatchSnapshot();
```

Comment:
Is the snapshot needed with all the assertions?

Reply:
Wanted to check the ordering of the content/icon and make sure no additional elements popped up (negative assertion, so to speak)